### PR TITLE
ENH: ensure id in index

### LINKED ISSUE #97: ID should consistently be used as an index for trackintel datatypes
All trackintel objects should have their unique identifier as the geopandas index. This should also be preserved if we process data. The reason we agreed on this was that pandas dataframes can lead to unexpected behavior if the index does not correspond to the ID (e.g., when merging or joining). 

This has to be considered at several places. At the moment I can think of the documentation, the accessors, i/o and functions that generate the outputs (e.g., that generate trips). Once it is implemented as a check in the accessors we will probably find most of the other places were we have to take care of this with the help of failing tests.

In the [documentation](https://trackintel.readthedocs.io/en/latest/modules/model.html ), we have already included the phrase _"the index of the GeoDataFrame will be treated as unique identifier of the Positionfixes"_ for all data types except for `Users`.

### LINKED ISSUE #119: Ensure generated id starts from 0
`Agg_level='user'` option in `generate_locations` function generates `id` starting from 1, which is not consistent with `Agg_level='dataset'`. Need to have ids starting from 0.
Tests for this issue are also needed.

### PR DESCRIPTION
**Pull request by hong2223**
closes #97.

- Test data files are changed accordingly.
- Tests are changed to accommodate the newly added warnings.
- New tests are added for the warnings.

Line 92 in `trackintel/preprocessing/staypoints.py` partly deals with #119

### CODE REVIEW

**Comment by henrymartin1** [comment]
Ah, I saw this PR just now. Is it possible to postpone everything that concerns triplegs until I finished #89?
This would mainly concern these two files: 
-  tests/preprocessing/test_positionfixes.py (not so important, but tests change a lot)
-  trackintel/preprocessing/positionfixes.py  (anything in generate_triplegs will be impossible to merge)

I guess we can also decide once I finished the PR...

**Comment by hong2223** [comment]
> Ah, I saw this PR just now. Is it possible to postpone everything that concerns triplegs until I finished #89?
> This would mainly concern these two files:
> 
> * tests/preprocessing/test_positionfixes.py (not so important, but tests change a lot)
> * trackintel/preprocessing/positionfixes.py  (anything in generate_triplegs will be impossible to merge)
> 
> I guess we can also decide once I finished the PR...

Sure, the changes concerning the two files are really small and I can easily integrate them afterwards

**Comment by hong2223** [comment]
I merged the latest pull request. The large number of changed files (sorry..) is mainly to avoid test warnings.

File: trackintel/preprocessing/staypoints.py [hunk 1]
```diff
@@ -105,7 +105,7 @@ def generate_locations(staypoints,
             
             ret_stps['location_id'] = labels
             
-        # create locations as grouped staypoints
+        ### create locations as grouped staypoints
```
Comment (henrymartin1) [inline trackintel/preprocessing/staypoints.py:108]: why '###' ?

**Review by henrymartin1 — APPROVED** [review]
Nice work! One step closer towards consistency ;-)

**Pull request merged**